stl_text/datamodule/doc_classification.py

```diff
@@ -27,17 +43,27 @@ def __init__(self, data_path: str = 'glue_sst2_tiny', vocab_path: Optional[str]
         self.text_transform = None
         self.label_transform = None
         self.datasets = {}
+        self.unk_token = '<unk>'
+
+    def setup(self, stage, tokenizer_type: str = 'sentencepiece'):
+        if tokenizer_type=='sentencepiece':
+            tokenizer = sentencepiece_tokenizer(download_from_url(PRETRAINED_SP_MODEL["text_unigram_25000"])).to_ivalue()
+            if self.vocab_path:
+                path_manager = PathManager()
+                with path_manager.open(self.vocab_path, "r",encoding='utf-8') as f:
+                    vocab = build_vocab_from_text_file(f,torch.jit.script(tokenizer))
+            else:
+                vocab = Vocab(VocabPybind([self.unk_token],self.unk_token))
```

Comment:
If `self.vocab_path` is not provided, we should check out the token->index dictionary in the sentencepiece model and use it in the vocab.